### PR TITLE
fix: fix react event name mappings BREAKING CHANGE

### LINKED ISSUE #296: Type Definitions for React CustomEvents
**Describe the bug**
EventHandler properties are not shown for CustomEvents on React & TypeScript

**To Reproduce**
1. Create `<BlSelect />` component
2. Add `onSelect`
3. On `onSelect={e => ...}` you won't be able to see custom properties of Event Handler referenced as **"e"** however it can be accessed

**Versions**
Baklava v2.0.0-beta.38
React 18.2.0

### LINKED ISSUE #307: Export component types
**Is your feature request related to a problem? Please describe.**
Sometimes when creating a custom component using existing Baklava component you could spread props. It would be great to import types of e.g "BlInputType" so we could use it as a type and spread. I'm open for suggestions if any :)

**Describe the solution you'd like**
Export `BlInputType` and other component types from `baklava-react.d.ts`

### LINKED ISSUE #397: Baklava identifies itself as Unknown in React
**Describe the bug**
Baklava not work in React 16. It identifies itself as "Unknown" component.
Possible: React Lazy Exports

**To Reproduce**
1. Upgrade baklava version to 2.0.0-beta.59 from 2.0.0-beta.48

**Screenshots**
Screen Shot 2023-02-02 at 11 01 12

**Desktop (please complete the following information):**
 - OS: Mac
 - Browser [e.g. chrome, safari]: Firefox
 - Version [e.g. 22]: 59

### PR DESCRIPTION
**Pull request by muratcorlu**
While debugging #397 with @Enes5519 we realized that some of our event names in react wrappers uses reserved event names from DOM events (like, `onClick`, `onInput`, `onChange`). And apparently that causes to have some incompatibility issues in some cases. This PR proposes to use regular event names those defined in our web components by just converting them to PascalCase and adding `on` as prefix.

Also, we were dropping some parts of the event names in react event mapping. Like `bl-dropdown-item-click` was `onClick` in current implementation. That was also potentially causing confusion.

This will need to update some projects those already uses Baklava in React. But since we are just about releasing our first stable release, I think it's still a good time to do so. Please give your comments if you use Baklava in React and if this new naming convention is future proof enough in your opinion.

This PR also fixes #296 by adding type definitions for events.

PS: This doesn’t break anything for web component implementations or using it in Vue. Only effects people who use Baklava in React. 

Here is a chart to compare old and new names:

| Component | Native Event Name | Current React Event | Proposed React Event |
| --- | --- | --- | --- |
| `BlAlert` | `bl-close` | `onClose` | `onBlClose`|
| `BlButton` | `bl-click` | `onClick` | `onBlClick`|
| `BlCheckboxGroup` | `bl-checkbox-group-change` | `onChange` | `onBlCheckboxGroupChange` |
| `BlDialog` | `bl-dialog-open` | `onOpen` | `onBlDialogOpen` |
| `BlDialog` | `bl-dialog-close` | `onClose` | `onBlDialogClose` |
| `BlDrawer` | `bl-drawer-open` | `onOpen` | `onBlDrawerOpen` |
| `BlDrawer` | `bl-drawer-close` | `onClose` | `onBlDrawerClose` |
| `BlDropdown` | `bl-dropdown-open` | `onOpen` | `onBlDropdownOpen` |
| `BlDropdown` | `bl-dropdown-close` | `onClose` | `onBlDropdownClose` |
| `BlIcon` | `bl-load` | `onLoad` | `onBlLoad` |
| `BlIcon` | `bl-error` | `onError` | `onBlError` |
| `BlInput` | `bl-change` | `onChange` | `onBlChange` |
| `BlInput` | `bl-input` | `onInput` | `onBlInput` |
| `BlInput` | `bl-invalid` | `onInvalid` | `onBlInvalid` |
| `BlPagination` | `bl-change` | `onChange` | `onBlChange` |
| `BlRadioGroup` | `bl-radio-change` | `onChange` | `onBlRadioChange` |
| `BlSelect` | `bl-select` | `onSelect` | `onBlSelect` |
| `BlSwitch` | `bl-switch-toggle` | `onToggle` | `onBlSwitchToggle` |
| `BlTextarea` | `bl-input` | `onInput` | `onBlInput` |
| `BlTextarea` | `bl-change` | `onChange` | `onBlChange` |
| `BlTextarea` | `bl-invalid` | `onInvalid` | `onBlInvalid` |
| `BlTooltip` | `bl-tooltip-show` | `onShow` | `onBlTooltipShow` |
| `BlTooltip` | `bl-tooltip-hide` | `onHide` | `onBlTooltipHide` |
| `BlCheckbox` | `bl-checkbox-change` | `onChange` | `onBlCheckboxChange` |
| `BlCheckbox` | `bl-focus` | `onFous` | `onBlFocus` |
| `BlCheckbox` | `bl-blur` | `onBlur` | `onBlBlur` |
| `BlDropdownItem` | `bl-dropdown-item-click` | `onClick` | `onBlDropdownItemClick` |
| `BlRadio` | `bl-checked` | `onChecked` | `onBlChecked` |
| `BlRadio` | `bl-focus` | `onFocus` | `onBlFocus` |
| `BlRadio` | `bl-blur` | `onBlur` | `onBlBlur` |
| `BlSelectOption` | `bl-select-option` | `onOption` | `onBlSelectOption` |
| `BlTab` | `bl-tab-selected` | `onSelected` | `onBlTabSelected` |

Fixes #397 
Fixes #307

### CODE REVIEW

**Comment by ozkersemih** [comment]
I think pr seems ok. Good job :)

**Comment by AykutSarac** [comment]
> PS: This doesn’t break anything for web component implementations or using it in Vue. Only effects people who use Baklava in React.

I'd recommend allowing previous events with the `@deprecated` flag, otherwise this change will likely to be adopted very late by the shell app has microfrontends and this will result in Baklava getting less feedbacks from community.

**Comment by muratcorlu** [comment]
> I'd recommend allowing previous events with the `@deprecated` flag, otherwise this change will likely to be adopted very late by the shell app has microfrontends and this will result in Baklava getting less feedbacks from community.

Because of we fetch our web components from CDN and React wrappers from NPM, increasing the CDN version in app shell, will not effect React events automatically. Every React app will need to update their NPM dependency separate. So I think the only risk is the pople that uses latest beta version in their package.json (which we suggest to avoid).

File: custom-elements-manifest.config.mjs [hunk 1]
```diff
@@ -27,7 +27,7 @@ export default {
 
                 return {
                   type: {
-                    text: `CustomEvent<${type.text.replace(/.*<([a-z]+)>/, '$1')}>`
+                    text: `CustomEvent<${type.text.replace('EventDispatcher<', '').replace(/>$/, '')}>`
```
Comment (Enes5519) [inline custom-elements-manifest.config.mjs:30]: Regex kullanabiliriz:
```js
type.text.match(/EventDispatcher<(.*?)>/s)[1]
```

**Comment by Enes5519** [comment]
It's still bad in type, but it solves our problem.

**Review by Enes5519 — APPROVED** [review]

**Review by ozkersemih — APPROVED** [review]

File: scripts/generate-react-exports.js [hunk 2]
```diff
@@ -38,26 +41,31 @@ for (const module of customElementsModules) {
       }, {})
     : {};
 
+  const eventTypes = events
+    ? ', {' + events.reduce((acc, curr) => {
+        return `${acc}${acc ? `, ` : ''}${getReactEventName(curr.name)}: EventName<${curr.type.text}>`
+      }, '') + '}'
+    : '';
```
[the remark this answers is not in the document]
Comment (muratcorlu) [reply scripts/generate-react-exports.js:47]: Make sense. I'll check and push if this works. 👍🏻

**Review by AykutSarac — APPROVED** [review]

**Pull request merged**